test/RoaveTest/ComposerGpgVerify/VerifyTest.php

```diff
@@ -0,0 +1,638 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RoaveTest\ComposerGpgVerify;
+
+use Composer\Composer;
+use Composer\Config;
+use Composer\Installer\InstallationManager;
+use Composer\Package\PackageInterface;
+use Composer\Repository\RepositoryInterface;
+use Composer\Repository\RepositoryManager;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+use PHPUnit\Framework\TestCase;
+use Roave\ComposerGpgVerify\Verify;
+use Symfony\Component\Process\Process;
+
+/**
+ * @covers \Roave\ComposerGpgVerify\Verify
+ */
+final class VerifyTest extends TestCase
+{
+    /**
+     * @var Event|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $event;
+
+    /**
+     * @var Composer|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $composer;
+
+    /**
+     * @var Config|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $config;
+
+    /**
+     * @var RepositoryManager|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $repositoryManager;
+
+    /**
+     * @var InstallationManager|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $installationManager;
+
+    /**
+     * @var RepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $localRepository;
+//$composer->getRepositoryManager()->getLocalRepository()->getPackages();
+//    //$composer->getLocker()->getLockData()
+//$composer->getInstallationManager()->getInstallPath();
+
+    /**
+     * @var string
+     */
+    private $originalGpgHome;
+
+    /**
+     * @var string
+     */
+    private $originalLanguage;
+
+    /**
+     * @var PackageInterface[] indexed by installation path
+     */
+    private $installedPackages = [];
+
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $this->installedPackages = [];
```

Comment:
Needs to be done to reset state at each run - there is no guarantee by the testing framework that the test will be instantiated each time.